src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/AvoidUnusedPrivateFieldsTests.cs

```diff
@@ -109,6 +212,76 @@ End Class
             GetCA1823BasicResultAt(7, 13, "Unused2"));
         }
 
+        [Fact, WorkItem(1219, "https://github.com/dotnet/roslyn-analyzers/issues/1219")]
+        public void CA1823_VisualBasic_FieldOffsetAttribute_NoDiagnostic()
+        {
+            VerifyBasic(@"
+<System.Runtime.InteropServices.StructLayout(System.Runtime.InteropServices.LayoutKind.Explicit)> _
+Public Class [Class]
+    <System.Runtime.InteropServices.FieldOffsetAttribute(8)> _
+    Private fieldWithFieldOffsetAttribute As Integer
+End Class
+");
+        }
+
+        [Fact, WorkItem(1219, "https://github.com/dotnet/roslyn-analyzers/issues/1219")]
+        public void CA1823_VisualBasic_FieldOffsetAttributeError_NoDiagnostic()
+        {
+            VerifyBasic(@"
+<System.Runtime.InteropServices.StructLayout(System.Runtime.InteropServices.LayoutKind.Explicit)> _
+Public Class [Class]
+    <System.Runtime.InteropServices.FieldOffsetAttribute> _
+    Private fieldWithFieldOffsetAttribute As Integer
+End Class
+", TestValidationMode.AllowCompileErrors);
+        }
+
+        [Fact, WorkItem(1217, "https://github.com/dotnet/roslyn-analyzers/issues/1217")]
+        public void CA1823_VisualBasic_MEFAttributes_NoDiagnostic()
+        {
+            VerifyBasic(BasicMEFAttributesDefinition + @"
+Public Class [Class]
+    <System.Composition.ExportAttribute> _
+    Private fieldWithMefV1ExportAttribute As Integer
+
+    <System.ComponentModel.Composition.ExportAttribute> _
+    Private fieldWithMefV2ExportAttribute As Integer
+End Class
+");
+        }
+
+        [Fact, WorkItem(1217, "https://github.com/dotnet/roslyn-analyzers/issues/1217")]
+        public void CA1823_VisualBasic_MEFAttributesError_NoDiagnostic()
+        {
+            VerifyBasic(BasicMEFAttributesDefinition + @"
+Public Class [Class]
+    <System.Composition.ExportAttribute(0)> _
+    Private fieldWithMefV1ExportAttribute As Integer
+
+    <System.ComponentModel.Composition.ExportAttribute(0)> _
+    Private fieldWithMefV2ExportAttribute As Integer
+End Class
+", TestValidationMode.AllowCompileErrors);
+        }
+
+        [Fact, WorkItem(1217, "https://github.com/dotnet/roslyn-analyzers/issues/1217")]
+        public void CA1823_VisualBasic_MEFAttributesUndefined_Diagnostic()
```

Comment:
No versions of this test for FieldOffsetAttribute?

Reply:
FieldOffsetAttribute is defined in mscorlib.